pkg/gadgettracermanager/containerutils/docker/docker.go

```diff
@@ -73,5 +73,9 @@ func (c *DockerClient) PidFromContainerId(containerID string) (int, error) {
 		return -1, fmt.Errorf("Container state is nil")
 	}
 
+	if containerJson.State.Pid == 0 {
+		return -1, fmt.Errorf("Pid for container %s is zero", containerID)
+	}
+
```

Comment:
Does it only happens in Docker? What about doing this check one level up on the stack so it works for all container runtimes?

Reply:
I have not tested with others, that might be the case too...

Reply:
It seems cri-o and containerd both [query the `State`](https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#query-state) from the OCI Container Runtime. And the [State](https://github.com/opencontainers/runtime-spec/blob/master/runtime.md#state) documentation says:

> pid (int, REQUIRED when status is `created` or `running` on Linux, OPTIONAL on other platforms) is the ID of the container process.

So it can be zero if the status is still `creating` or already `stopped`. So I will move the check one level up on the stack as you suggested.

Reply:
Nice to know. Thanks!